composition-js/src/merging/merge.ts

```diff
@@ -1662,7 +1662,7 @@ class Merger {
     }
     
     // if there is a @fromContext directive on one of the source's arguments, we need a join__field
-    if (sources.some((s, idx) => {
```

Comment:
Unfortunately, this will introduce a bug because `idx` is the subgraph index which will be incorrect if the `undefined`s are pruned from the original array.

Reply:
ah good catch, I initially had at the beginning of the closure: 
```typescript
if (!s) {
  return false;
}
``` but changed it to use `filter` thinking it was equivalent 🤦